### DISCUSSION TITLE
Pictures with different color styles

### DISCUSSION THREAD

**Discussion by lightSpeedBubble** [hsoft/dupeguru#1122] [Ideas]
I found a solution to the problem of Dupeguru not finding the same picture with different color styles (color, sepia, greyscale) or even drawings at different stages of progress (sketch, line art, flat colors, shaded colors). It involves creating a copy of the pictures folder that contains copies of all the images, filtered to have only the contour lines and an added thickness, and scanning that copy instead.
I made [a script](https://gitlab.com/LightSpeedBubble/image-organizing-tools/-/blob/main/contourer/contourer-README.md) for it and it works very well. Dupeguru can now find similar drawings it could not find before.
That said, it would be better if there was natively an option in Dupeguru's Picture mode to look for such pictures.

**Comment by glubsy** [comment]
This is interesting, but only a workaround. 

If we really want a solution to the issue here, I suggest having a look at the [actual code](https://github.com/arsenetar/dupeguru/tree/master/core/pe/modules).

**Comment by lightSpeedBubble** [comment]
Unfortunately I don't know C and I'm not familiar with image similarity algorithms so I don't understand the code in these files. My workaround is a simple bash script, far from an optimal solution.

I assume a real one would involve providing an option in Dupeguru (as a checkbox) to also look for similarity in contour lines, which, if superior to the regular similarity, would overtake its matching value.

Similarity between a line art and colorized drawing can be close to 0% while their contour lines can be close to 100%, so I don't think the 2 values should be averaged or multiplied because it would still drown the image couple in a pool of completely unrelated images. That's why I suggest overtaking the value.

**Comment by rkazak** [comment]
It could also be done if Dupeguru allowed executing a pre-command against the files being compared. Maybe thru the same or similar mechanism  used for custom command?
